eng/common/scripts/Invoke-Cspell.ps1

```diff
@@ -0,0 +1,163 @@
+<#
+.SYNOPSIS
+Invokes a given version of cspell with standard parameters and provided globs
+
+.PARAMETER JobType
+Maps to cspell command (e.g. `lint`, `trace`, etc.). Default is `lint`
+
+.PARAMETER ScanGlobs
+List of glob expressions to be scanned. This list is not constrained by
+npx/cmd's upper limit on command line length as the globs are inserted into the
+cspell config's `files` property.
+
+.PARAMETER Packages
+Array of strings of the form "<package>@<version>" that will be used in the npx
+run. Example: @("cspell@12.5.6", "@cspell/cspell-bundled-dicts@12.5.3"). Default
+is a basic set of packages that work together.
+
+.PARAMETER CSpellConfigPath
+Location of cspell.json file to use when scanning. Defaults to
+`.vscode/cspell.json` at the root of the repo.
+
+.PARAMETER SpellCheckRoot
+Location of root folder for generating readable relative file paths. Defaults to
+the root of the repo relative to the script.
+
+.PARAMETER Test
+Run test functions against the script logic
+
+.EXAMPLE
+./eng/common/scripts/Invoke-Cspell.ps1 -ScanGlobs 'sdk/*/*/PublicAPI/**/*.md'
+
+This will run spell check with the given globs
+
+.EXAMPLE
+./eng/common/scripts/Invoke-Cspell.ps1 -ScanGlobs @('sdk/storage/**', 'sdk/keyvault/**')
+
+This will run spell check against multiple globs
+
+.EXAMPLE
+./eng/common/scripts/Invoke-Cspell.ps1 -ScanGlobs './README.md'
+
+This will run spell check against a single file
+
+#>
+[CmdletBinding()]
+param(
+  [Parameter()]
+  [string] $JobType = 'lint',
+
+  [Parameter()]
+  [array]$ScanGlobs = '**',
+
+  [Parameter()]
+  [array]$CSpellPackages = @(
+    "cspell@5.12.3",
```

Comment:
Do we really need to specify all of these?

Reply:
Tried a new approach using `npm install` and `npm exec`